### PR TITLE
NXP-32275: admin-console-confirmation-request-dialog

### PR DESCRIPTION
**Pull request by Nishant0928**
https://jira.nuxeo.com/browse/NXP-32275

### CODE REVIEW

File: nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/nxql-es-reindex/nxql-es-reindex.component.scss [hunk 1]
```diff
@@ -10,3 +10,20 @@
...
+  width: 100%;
+  height: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-color: rgba(255, 255, 255, 0.7);
+  z-index: 9999;
+}
+
+.custom-spinner {
```
Comment (swarnadipa-dev) [inline nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/nxql-es-reindex/nxql-es-reindex.component.scss:26]: Are we using this class ?

File: nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/folder-es-reindex/folder-es-reindex.component.ts [hunk 2]
```diff
@@ -320,4 +327,42 @@ export class FolderESReindexComponent implements OnInit, OnDestroy {
     this.launchedDialogClosedSubscription?.unsubscribe();
     this.errorDialogClosedSubscription?.unsubscribe();
   }
-}
\ No newline at end of file
+}
+
+function secondsToHumanReadable(seconds: number): string {
```
Comment (swarnadipa-dev) [inline nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/folder-es-reindex/folder-es-reindex.component.ts:332]: This function should be moved to a pipe or a service & reused.

File: nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/nxql-es-reindex/nxql-es-reindex.component.html [hunk 3]
```diff
@@ -1,15 +1,35 @@
-<form [formGroup]="nxqlReindexForm" (ngSubmit)="onReindexFormSubmit()" class="nxql-es-reindex-container">
-    <mat-form-field hyFormField>
-        <mat-label>{{ ELASTIC_SEARCH_LABELS.NXQL_QUERY }}</mat-label>
-        <textarea matInput class="nxql-query-textarea" [placeholder]="ELASTIC_SEARCH_LABELS.NXQL_QUERY_PLACEHOLDER"
-            formControlName="nxqlQuery" id="nxqlQuery" required></textarea>
-        <mat-error *ngIf="nxqlReindexForm?.get('nxqlQuery')?.invalid">{{
-            getErrorMessage()
-            }}</mat-error>
-    </mat-form-field>
+<form
```
Comment (swarnadipa-dev) [inline nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/nxql-es-reindex/nxql-es-reindex.component.html:1]: please format this file

File: nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/folder-es-reindex/folder-es-reindex.component.html [hunk 4]
```diff
@@ -1,12 +1,33 @@
...
-        <mat-error *ngIf="folderReindexForm?.get('documentID')?.invalid">
-            {{getErrorMessage()}}
-        </mat-error>
-    </mat-form-field>
-    <button mat-flat-button color="primary" type="submit" class="folder-es-reindex-container__button">
-        {{ ELASTIC_SEARCH_LABELS.REINDEX_BUTTON_LABEL }}
-    </button>
-</form>
\ No newline at end of file
+<form
```
Comment (swarnadipa-dev) [inline nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/folder-es-reindex/folder-es-reindex.component.html:1]: please format this file

File: nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/services/elastic-search-reindex.service.ts [hunk 5]
```diff
@@ -1,42 +1,60 @@
...
+  }
+
   performDocumentReindex(requestQuery: string | null): Observable<ReindexInfo> {
-    return this.http.post<ReindexInfo>(
-      `${this.nuxeoJsClientService.getApiUrl()}${
-        this.elaticSearchReindexEndpoint
-      }?query=${requestQuery}`,
-      {}
-    );
+    this.setSpinnerStatus(true);
```
Comment (swarnadipa-dev) [inline nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/services/elastic-search-reindex.service.ts:26]: this should move to the point where this action is dispatched, in the component.

**Review by swarnadipa-dev — CHANGES_REQUESTED** [review]

File: nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/nxql-es-reindex/nxql-es-reindex.component.scss (hunk 1 above)
Comment (Nishant0928) [reply nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/nxql-es-reindex/nxql-es-reindex.component.scss:26]: yes we are using it

**Review by swarnadipa-dev — APPROVED** [review]

File: nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/folder-es-reindex/folder-es-reindex.component.html [hunk 6]
```diff
@@ -9,4 +9,11 @@
     <button mat-flat-button color="primary" type="submit" class="folder-es-reindex-container__button">
         {{ ELASTIC_SEARCH_LABELS.REINDEX_BUTTON_LABEL }}
     </button>
-</form>
\ No newline at end of file
+    <div class="spinner-overlay" *ngIf="spinnerVisible">
+        <mat-progress-spinner class="custom-spinner" mode="indeterminate">
```
Comment (rakeshkumar1019) [inline nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/folder-es-reindex/folder-es-reindex.component.html:13]: <mat-progress-spinner `hyIcon` class="custom-spinner" mode="indeterminate">
 Ref:https://ui.hyland.io/5932dab8d/v/41976/p/0716ad-icon/t/486172

File: nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/nxql-es-reindex/nxql-es-reindex.component.html [hunk 7]
```diff
@@ -12,4 +12,10 @@
         {{ ELASTIC_SEARCH_LABELS.REINDEX_BUTTON_LABEL }}
     </button>
     <div [innerHTML]="nxqlQueryHintSanitized"></div>
-</form>
\ No newline at end of file
+    <div class="spinner-overlay" *ngIf="spinnerVisible">
+        <mat-progress-spinner class="custom-spinner" mode="indeterminate">
+        </mat-progress-spinner>
```
Comment (rakeshkumar1019) [inline nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/components/nxql-es-reindex/nxql-es-reindex.component.html:17]: add `hyIcon` Directive here too

**Review by rakeshkumar1019 — CHANGES_REQUESTED** [review]

**Review by madhurkulshrestha-hyland — APPROVED** [review]

**Review by rakeshkumar1019 — APPROVED** [review]

**Review by rakeshkumar1019 — APPROVED** [review]

**Pull request merged**